### PR TITLE
BrexitPrimary section

### PR DESCRIPTION
**Pull request by callumlocke**
This section is behind the flags `frontPageBrexitPrimaryUK` and `frontPageBrexitPrimaryInternational`. We're planning to activate one or both of these on Wednesday or Thursday, and disable them after the referendum result is called.

NB. this PR also includes some code for an unfinished sub-feature, `BrexitPrimaryInteractive` – a React component showing live data, which sits inside the `BrexitPrimary` section. This component is behind its own flag `frontPageBrexitPrimaryInteractive`, and it needs more work, but we want to merge this code in now for stakeholders' review. (Not visible in screenshot below.)

image

### CODE REVIEW

**Comment by callumlocke** [comment]
@ironsidevsquincy and @andygout, please review

File: client/main.js [hunk 1]
```diff
@@ -6,13 +7,26 @@ import highlightDomPath from './components/highlight-dom-path/highlight-dom-path
...
 	highlightDomPath();
+	myft(flags);
+	setupBrexitPrimaryInteractive();
+
+	let config = {
+		placeholder: true,
+		classes: ['video'],
+		selector: '.js-video',
+		advertising: flags.get('videoPlayerAdvertising')
+	};
```
Comment (ironsidevsquincy) [inline client/main.js]: Duplicated? (see later in file)

File: config/layouts/brexit-primary.js [hunk 2]
```diff
@@ -1,27 +1,21 @@
...
+				colspan: { default: 12 },
 				components: [
 					{
 						type: 'Content',
 						size: 'large-no-image',
 						standfirst: {
 							show: { default: true }
 						},
-						lazyLoadImages: flags.frontPageLazyLoadImages
+						// lazyLoadImages: flags.frontPageLazyLoadImages
```
Comment (ironsidevsquincy) [inline config/layouts/brexit-primary.js]: might as well delete

File: shared/components/brexit-primary-interactive/_brexit-primary-interactive.scss [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+.brexit-primary-interactive {
+	// outline: 1px solid red;
+	color: white;
+
+	width: 100%;
+	position: relative;
+}
+	.brexit-primary-interactive__progress-control {
```
Comment (ironsidevsquincy) [inline shared/components/brexit-primary-interactive/_brexit-primary-interactive.scss]: indentation

File: package.json [hunk 4]
```diff
@@ -18,6 +18,7 @@
     "babel-preset-react": "^6.3.13",
     "babel-register": "^6.4.3",
     "body-parser": "^1.13.2",
+    "classnames": "^2.2.5",
```
Comment (ironsidevsquincy) [inline package.json]: Think we have something that pretty much does this

https://github.com/Financial-Times/n-card/blob/master/src/lib/helpers.js#L54
e.g. https://github.com/Financial-Times/n-card/blob/master/components/article.js#L100

Could export the helpers from `n-card` and use it instead

File: client/main.js (hunk 1 above)
Comment (callumlocke) [inline client/main.js]: whoops, bad conflict resolution

**Comment by callumlocke** [comment]
thanks @ironsidevsquincy, could you take another look?

**Comment by ironsidevsquincy** [comment]
👍

**Pull request merged**